src/lib/schemas/schemaStore.tsx

```diff
@@ -25,6 +26,27 @@ export const useSchemaStore = create<SchemasStore>()(
     }))
 )
 
+/*
+    Separating the "vanilla" store from the bound tore hook to make testing easier
+    based on ideas here - https://docs.pmnd.rs/zustand/guides/typescript#bounded-usestore-hook-for-vanilla-stores
+*/
+const createBoundedUseStore = ((store) => (selector, equals) =>
+    useStore(store, selector as never, equals)) as <
+    S extends StoreApi<unknown>
+>(
+    store: S
+) => {
+    (): ExtractState<S>
+    <T>(
+        selector: (state: ExtractState<S>) => T,
+        equals?: (a: T, b: T) => boolean
+    ): T
+}
```

Comment:
I was just looking at this for another PR. Do we need this? You can just call `setState` on the hook? 

```ts
useSchemaStore.setState()
```

Currently these are not exported from the index in eg. `/schemas`, maybe we should do that as well - or do we not want it as part of the "public" API, but ok to just import directly in tests?

Reply:
See here: https://github.com/dhis2/maintenance-app-beta/pull/347/files#diff-47488971ff4a34452107ca225072da66e8f752f892d326c96004e0fbd1dee5afR159